test/hierarchical/test_ordinal.py

```diff
@@ -0,0 +1,310 @@
+    petab_problem = petab.Problem.from_yaml(example_ordinal_yaml)
+    vals = []
+    for idx, option in enumerate(inner_solver_options):
+        problem = _create_problem(petab_problem, option)
+        val = problem.objective(np.array([0, 0]))
+        vals.append(val)
+        assert np.isclose(vals[idx], vals[idx - 1])
+
+
+def test_optimization(inner_solver_options: List[Dict]):
+    """Check that optimizations finishes without error."""
+    petab_problem = petab.Problem.from_yaml(example_ordinal_yaml)
+
+    optimizer = pypesto.optimize.ScipyOptimizer(
+        method='Nelder-Mead', options={'maxiter': 10}
+    )
+    for option in inner_solver_options:
+        problem = _create_problem(petab_problem, option)
+        pypesto.optimize.minimize(
+            problem=problem, n_starts=1, optimizer=optimizer
+        )
+
+
+def _create_problem(
+    petab_problem: petab.Problem, option: Dict
+) -> pypesto.Problem:
+    """Creates the ordinal pyPESTO problem with given options."""
+    importer = pypesto.petab.PetabImporter(petab_problem, ordinal=True)
+    importer.create_model()
+
+    objective = importer.create_objective(
+        inner_solver_options=option,
+    )
+    problem = importer.create_problem(objective)
+    return problem
+
+
+def test_optimal_scaling_calculator_and_objective():
+    """Test the optimal scaling calculation of objective and gradient values."""
+    petab_problem = petab.Problem.from_yaml(example_ordinal_yaml)
+
+    methods = [STANDARD, REDUCED]
+
+    options_per_method = {
+        STANDARD: {'method': STANDARD, 'reparameterized': False},
+        REDUCED: {'method': REDUCED, 'reparameterized': False},
+    }
+    problems = {}
+
+    for method, options in options_per_method.items():
+        importer = pypesto.petab.PetabImporter(petab_problem, ordinal=True)
+        objective = importer.create_objective(
+            inner_solver_options=options,
+        )
+        problem = importer.create_problem(objective)
+        problems[method] = problem
+
+    def calculate(problem, x_dct):
+        return problem.objective.calculator(
+            x_dct=x_dct,
+            sensi_orders=(0, 1),
+            mode=MODE_FUN,
+            amici_model=problem.objective.amici_model,
+            amici_solver=problem.objective.amici_solver,
+            edatas=problem.objective.edatas,
+            n_threads=1,
+            x_ids=petab_problem.x_ids,
+            parameter_mapping=problem.objective.parameter_mapping,
+            fim_for_hess=False,
+        )
+
+    x_dct = dict(zip(petab_problem.x_ids, petab_problem.x_nominal_scaled))
+
+    calculator_results = {
+        method: calculate(problems[method], x_dct=x_dct) for method in methods
+    }
+
+    # The results of the objective gradient and function value
+    # should not depend on the method given.
+    assert np.isclose(
+        calculator_results[STANDARD]['fval'],
+        calculator_results[REDUCED]['fval'],
+    )
+    assert np.isclose(
```

Comment:
could you also compare gradients to finite differences? --> pypesto.objective.FD